src/App/WarehouseService.cs

```diff
@@ -3,25 +3,45 @@
 
 public class WarehouseService
 {
-    private readonly string _connectionString = "Host=localhost;Username=postgres;Password=yourpassword;Database=warehouse";
+    private readonly string _connectionString = Environment.GetEnvironmentVariable("CONNECTION_STRING");
 
-    public void SaveData(string key, string value)
+    public void SaveData(string partName, int bikeTypeId, int stock)
     {
+        try
+        {
+            using var connection = new NpgsqlConnection(_connectionString);
+            connection.Open();
 
-        try {
-        using var connection = new NpgsqlConnection(_connectionString);
-        connection.Open();
+            // Step 1: Check if the BikeType exists in the Bike_Types table.
+            using var checkBikeTypeCmd = new NpgsqlCommand("SELECT \"BikeTypeID\" FROM \"Bike_Types\" WHERE \"BikeTypeID\" = @bikeTypeId", connection);
+            checkBikeTypeCmd.Parameters.AddWithValue("bikeTypeId", bikeTypeId);
+            var bikeTypeExists = checkBikeTypeCmd.ExecuteScalar() != null;
 
-        using var cmd = new NpgsqlCommand("INSERT INTO data (key, value) VALUES (@k, @v)", connection);
-        cmd.Parameters.AddWithValue("k", key);
-        cmd.Parameters.AddWithValue("v", value);
-        cmd.ExecuteNonQuery();
+            if (!bikeTypeExists)
+            {
+                Console.WriteLine("Error: BikeTypeID does not exist.");
+                return;
+            }
 
-        Console.WriteLine($"Data saved to Warehouse: Key={key}, Value={value}");
+            // Step 2: Insert the part into the Bike_Parts table.
+            using var cmdInsertPart = new NpgsqlCommand("INSERT INTO \"Bike_Parts\" (\"PartName\", \"BikeTypeID\", \"Stock\") VALUES (@partName, @bikeTypeId, @stock)", connection);
```

Comment:
```suggestion
      using var cmdInsertPart = new NpgsqlCommand(@"INSERT INTO Bike_Parts (PartName, BikeTypeID, Stock) 
                                             VALUES (@partName, @bikeTypeId, @stock)", connection);
```

Reply:
String literals can easily become a parsing nightmare and can be hard to spot, so this is for improved maintainablity and debugging ability.